app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt

```diff
@@ -469,13 +462,14 @@ class SettingsFragment constructor(
             .commit()
     }
 
-    private fun authenticationResult(result: Int) {
+    private fun setLockAuthenticationResult(result: Int): Boolean {
         val success = result == Authenticator.SUCCESS
         val switchLock = findPreference<SwitchPreference>("app_lock")
         switchLock?.isChecked = success
 
         findPreference<SwitchPreference>("app_lock_home_bypass")?.isVisible = success
         findPreference<EditTextPreference>("session_timeout")?.isVisible = success
+        return success
```

Comment:
What is the point of returning `success` here, as it is basically the input and can also be determined outside the function?

Reply:
This function is passed as a callback to the authentication handling performed in the settingsActivity. `succes` indicates if the authentication flow is completely handled, and if so the callback is removed. Since the authenticator in the settingsActivity now has to fulfill both the app locking and the original functionality from the settingsFragment, the return of success is used for the bookkeeping which authentication mode should be active. (See the use of this return value in the settingsActivity)

Reply:
But if the result is not `Authenticator.SUCCESS`, you would still handle the response in this function (by changing the setting to off) while the callback won't be removed. Is that correct?

Reply:
You are correct. When the authentication is canceled the callback shoul also be deleted.